cpp/src/slice2java/JavaUtil.cpp

```diff
@@ -1720,6 +1722,16 @@ Slice::JavaGenerator::validateMetadata(const UnitPtr& u)
         .acceptedArgumentKind = MetadataArgumentKind::SingleArgument,
         .extraValidation = [](const MetadataPtr&, const SyntaxTreeBasePtr& p) -> optional<string>
         {
+            // TODO uncomment after replacing 'java:package' with 'java:identifier'.
+            // const string msg = "'java:package' is deprecated; use 'java:identifier' to remap modules instead";
+            // p->unit()->warning(metadata->file(), metadata->line(), Deprecated, msg);
+
+            if (auto cont = dynamic_pointer_cast<Contained>(p); cont && cont->hasMetadata("java:identifier"))
+            {
+                return "the 'java:package' metadata cannot be used alongside 'java:identifier' - both change the name "
```

Comment:
Isn't java:package more of a prefix?

Reply:
See C# comment. And in Java, it's mapped package.